src/main/scala/MockedStreams.scala

```diff
@@ -30,24 +30,29 @@ object MockedStreams {
 
   def apply() = Builder()
 
-  case class Input(seq: Seq[(Array[Byte], Array[Byte])])
+  case class Event(topic: String, key: Array[Byte], value: Array[Byte])
 
   case class Builder(topology: Option[(KStreamBuilder => Unit)] = None,
                      configuration: Properties = new Properties(),
                      stateStores: Seq[String] = Seq(),
-                     inputs: Map[String, Input] = Map()) {
+                     inputs: List[Event] = List.empty) {
+
 
     def config(configuration: Properties) = this.copy(configuration = configuration)
 
     def topology(func: (KStreamBuilder => Unit)) = this.copy(topology = Some(func))
 
     def stores(stores: Seq[String]) = this.copy(stateStores = stores)
 
-    def input[K, V](topic: String, key: Serde[K], value: Serde[V], seq: Seq[(K, V)]) = {
+    def input[K, V](topic: String, key: Serde[K], value: Serde[V], newInput: Seq[(K, V)]) = {
       val keySer = key.serializer
       val valSer = value.serializer
-      val in = seq.map { case (k, v) => (keySer.serialize(topic, k), valSer.serialize(topic, v)) }
-      this.copy(inputs = inputs + (topic -> Input(in)))
+
+      val updatedInputs = newInput.foldLeft(inputs) {
+        case (events, (k, v)) => Event(topic, keySer.serialize(topic, k), valSer.serialize(topic, v)) :: events
```

Comment:
events :: Event(...) then we don't need to reverse the inputs in line 107.

Reply:
I agree it would be more readable if we could avoid the `reverse` down below.

On the other side, appending elements at the beginning of the list here with `::` is the fastest possible: O(1) since it just adding an element to a linked list). Appending at the end of the List can be done with `++` though is slower as I understand it, [this blog](https://alvinalexander.com/scala/scala-list-class-examples#how-to-prepend-items-to-a-list) mentions some details about it. 

=> This is your call :) If you'd rather have the most readable code possible I'm happy to convert this into something like `events ++ List(Event(...)) `, or if you prefer performance we can keep it like this.